src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
+import { format, fromUnixTime, getUnixTime } from "date-fns";
+import { range } from "lodash";
+import { useCss } from "react-use";
+
+import { RoomWithEvents, ScheduleProps } from "./Schedule.types";
+
+import { calcStartPosition } from "./Schedule.utils";
+
+import { ScheduleRoomEvents } from "../ScheduleRoomEvents";
+
+import "./Schedule.scss";
+
+const MAX_SCHEDULE_START_HOUR = 16;
+const MAX_HOUR = 24;
+const MIDDAY_HOUR = 12;
+
+export const Schedule: React.FC<ScheduleProps> = ({ scheduleDay }) => {
+  const getStartHour = useCallback(
+    (utcSeconds: number) => {
+      return utcSeconds >= scheduleDay.dayStartUtcSeconds
+        ? Number(format(fromUnixTime(utcSeconds), "H"))
+        : 0;
+    },
+    [scheduleDay]
+  );
+
+  const roomStartHour = useCallback(
+    (room: RoomWithEvents) =>
+      room.events?.reduce(
+        (acc, event) => Math.min(acc, getStartHour(event.start_utc_seconds)),
+        MAX_SCHEDULE_START_HOUR
+      ) ?? MAX_SCHEDULE_START_HOUR,
+    [getStartHour]
+  );
+
+  const scheduleStartHour = useMemo(
+    () =>
+      Math.min(
+        ...scheduleDay.rooms.map((room) => roomStartHour(room)),
+        MAX_SCHEDULE_START_HOUR
+      ),
+    [scheduleDay, roomStartHour]
+  );
+
+  const hours = useMemo(
+    () =>
+      range(scheduleStartHour, MAX_HOUR).map(
+        (hour: number) =>
+          `${hour % MIDDAY_HOUR || MIDDAY_HOUR} ${
+            hour >= MIDDAY_HOUR ? "PM" : "AM"
+          }`
+      ),
+    [scheduleStartHour]
+  );
+
+  const containerVars = useCss({
+    "--room-count": scheduleDay.rooms.length + 1,
+    "--current-time--position": calcStartPosition(
+      Math.floor(getUnixTime(Date.now())),
+      scheduleStartHour
+    ),
+    "--hours-count": hours.length,
+  });
+
+  const containerClasses = useMemo(
+    () => classNames("ScheduledEvents", containerVars),
+    [containerVars]
+  );
+
+  return (
+    <div className={containerClasses}>
+      <div className="ScheduledEvents__rooms">
+        <div className="ScheduledEvents__room">
+          <p className="ScheduledEvents__room-title">My Daily Schedule</p>
+          <span className="ScheduledEvents__events-count">
+            {scheduleDay.personalEvents.length} events
+          </span>
+        </div>
+        {scheduleDay.rooms.map((room) => (
+          <div key={room.title} className="ScheduledEvents__room">
```

Comment:
the logic is changed a bit, now the key consists of 2 parts: `venueId` + `roomTitle`: https://github.com/sparkletown/sparkle/pull/1302/commits/28b4cc25e369c09ac8ce1a6b2e3f4ee709068144

Reply:
Can there be 2 rooms in the same venue with the same `roomTitle`? I don't think we do anything to prevent this when creating rooms/etc. So might be worth appending the `index` to this `key` as well. And maybe adding a comment about why we are using `index` there like we have in some of the other places in the codebase.

This is an overly verbose example:
```
// @debt Ideally event.id would always be a unique identifier, but our types suggest it
//   can be undefined. Because we can't use index as a key by itself (as that is unstable
//   and causes rendering issues, we construct a key that, while not guaranteed to be unique,
//   is far less likely to clash
```

Reply:
I group rooms earlier by `venueId` and `roomTitle`, so the pairs will be unique. 
And if there are 2 rooms in the same venue with the same `roomTItle` they will be in the same schedule line.

Reply:
Cool, makes sense. Could you maybe add a comment saying that so we know/remember in future.

And to https://github.com/sparkletown/sparkle/pull/1302#discussion_r633809377

Reply:
added a comment: https://github.com/sparkletown/sparkle/pull/1302/commits/03d78303b9d16dfcdfde3ed631a87b04eec082ea